test/users/index.spec.js

```diff
@@ -0,0 +1,52 @@
+const request = require('supertest');
+
+const app = require('../../app');
+const { createManyUsers } = require('../factory/users_factory');
+const { showUserSerializer } = require('../../app/serializers/users');
+
+describe('GET /users', () => {
+  const httpRequest = (query = {}) =>
+    request(app)
+      .get('/users')
+      .query(query);
+  let usersIndexResponse = {};
+  let users = {};
+  beforeAll(async () => {
+    users = await createManyUsers(2);
+  });
+
+  describe('without query params', () => {
+    beforeAll(async () => {
+      usersIndexResponse = await httpRequest();
+    });
+
+    test('Responds with 200 status code', () => expect(usersIndexResponse.statusCode).toBe(200));
+
+    test('Responds with the expected body structure', () =>
+      expect(usersIndexResponse.body).toMatchObject({
+        page: expect.any(Array),
+        total_count: expect.any(Number)
+      }));
```

Comment:
Remember that the `toHaveExpectedProperty` could be used here.

Reply:
I took a look of this, but with `toMatchObject` I can test the return types.

Reply:
Yes. But we can make that two validations in separate tests. If this test fails I don't know if the field is not present in the response or if it has a different type